mlir/lib/Conversion/PDLToPDLInterp/PDLToPDLInterp.cpp

```diff
@@ -151,7 +151,7 @@ struct PatternLowering {
 
   /// A mapping between constraint questions that refer to values created by
   /// constraints and the temporary placeholder values created for them.
-  DenseMap<std::pair<ConstraintQuestion *, unsigned>, Value> substitutions;
+  std::multimap<std::pair<ConstraintQuestion *, unsigned>, Value> substitutions;
```

Comment:
Can you use `unordered_multimap` instead?

Reply:
Yes I think this would be a good idea, but it does not work out of the box together with `std::pair` as it does not know how to hash it. 
We could add a custom hash function here, but to keep the changes brief I did not opt for this.